core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -370,10 +370,12 @@ class UnitMovementAlgorithms(val unit: MapUnit) {
      * Displace a unit - choose a viable tile close by if possible and 'teleport' the unit there.
      * This will not use movement points or check for a possible route.
      * It is used e.g. if an enemy city expands its borders, or trades or diplomacy change a unit's
-     * allowed position.
+     * allowed position. Does not move transported units by default but setting
+     * [forceMoveTransportedUnit] to true allows them to be moved.
      * CAN DESTROY THE UNIT.
      */
-    fun teleportToClosestMoveableTile() {
+    fun teleportToClosestMoveableTile(forceMoveTransportedUnit: Boolean = false) {
+        if (unit.isTransported && !forceMoveTransportedUnit) return // handled when carrying unit is teleported
```

Comment:
Seems very strange. The bool gates this early exit for the "planes" but not the pull-along code for the "carrier"? So the flag is only for the case you want to teleport a transported plane but leave the carrier? Would that even work correctly or leave isTransported invalid? I think we do want the "carrier pulls along" to be default, so city expansion or a modded carrier with withdraw chance, for example, would do it too...  OK, I may not understand everything here, but seems like a dead flag, not what the Kdoc seems to describe.

Reply:
Yeah that's a good point. I mainly added in case there was a future use that would want to teleport a transported unit separately from its transport but since I don't even know when this case would happen it's best to not implement it.